scanner/src/main/kotlin/provenance/NestedProvenanceScanResult.kt

```diff
@@ -40,16 +40,11 @@ data class NestedProvenanceScanResult(
      */
     val scanResults: Map<KnownProvenance, List<ScanResult>>
 ) {
-    /**
-     * Return a set of all [KnownProvenance]s contained in [nestedProvenance].
-     */
-    fun getProvenances(): Set<KnownProvenance> = nestedProvenance.getProvenances()
-
     /**
      * Return true if [scanResults] contains at least one scan result for each of the [KnownProvenance]s contained in
      * [nestedProvenance].
      */
-    fun isComplete(): Boolean = getProvenances().all { scanResults[it]?.isNotEmpty() == true }
+    fun isComplete(): Boolean = nestedProvenance.allProvenances.all { scanResults[it]?.isNotEmpty() == true }
```

Comment:
The removal of the function in line 46 seems to me strictly speaking a bit unrelated / a thing which could become a separate commit. However, would it make sense to keep this function to not introduce these Law of Demeter violations?

Reply:
I agree about maybe splitting this out to a separate commit, but how is the principle of least knowledge violated here?

Reply:
Just notice it's just once, the occurrence in NestedProvenanceScanResultTest.kt, so let's leave it as is.